ArcGISRuntimeSDKQt_CppSamples/Analysis/ViewshedGeoElement/ViewshedGeoElement.cpp

```diff
@@ -0,0 +1,184 @@
+// [Legal]
+
+#include "ViewshedGeoElement.h"
+
+#include "ArcGISTiledElevationSource.h"
+#include "Scene.h"
+#include "SceneQuickView.h"
+#include "Camera.h"
+#include "Point.h"
+#include "Viewpoint.h"
+#include "ArcGISSceneLayer.h"
+#include "SimpleRenderer.h"
+#include "ModelSceneSymbol.h"
+#include "GeoElementViewshed.h"
+#include "GeometryEngine.h"
+#include "OrbitGeoElementCameraController.h"
+
+#include <QQmlProperty>
+#include <QTimer>
+#include <QString>
+#include <QUrl>
+#include <QVariant>
+#include <QList>
+
+using namespace Esri::ArcGISRuntime;
+
+ViewshedGeoElement::ViewshedGeoElement(QQuickItem* parent /* = nullptr */):
+  QQuickItem(parent)
+{
+}
+
+void ViewshedGeoElement::init()
+{
+  // Register classes for QML
+  qmlRegisterType<SceneQuickView>("Esri.Samples", 1, 0, "SceneView");
+  qmlRegisterType<ViewshedGeoElement>("Esri.Samples", 1, 0, "ViewshedGeoElementSample");
+}
+
+void ViewshedGeoElement::componentComplete()
+{
+  QQuickItem::componentComplete();
+
+  // Create a scene and give it to the SceneView
+  m_sceneView = findChild<SceneQuickView*>("sceneView");
+  Scene* scene = new Scene(Basemap::imagery(this), this);
+
+  // Create a surface
+  Surface* surface = new Surface(this);
+  surface->elevationSources()->append(
+        new ArcGISTiledElevationSource(
+          QUrl("http://scene.arcgis.com/arcgis/rest/services/BREST_DTM_1M/ImageServer"),
+          this));
+  scene->setBaseSurface(surface);
+
+  // Add a SceneLayer
+  ArcGISSceneLayer* sceneLayer = new ArcGISSceneLayer(QUrl("http://tiles.arcgis.com/tiles/P3ePLMYs2RVChkJx/arcgis/rest/services/Buildings_Brest/SceneServer/layers/0"), this);
+  scene->operationalLayers()->append(sceneLayer);
+
+  // Add an AnalysisOverlay
+  m_analysisOverlay = new AnalysisOverlay(this);
+  m_sceneView->analysisOverlays()->append(m_analysisOverlay);
+
+  // Add a GraphicsOverlay
+  createGraphicsOverlay();
+
+  // Create a Graphic
+  createGraphic();
+
+  // Create the GeoElementViewshed
+  const double horizontalAngle = 90.0;
+  const double verticalAngle = 25.0;
+  const double minDistance = 1.0;
+  const double maxDistance = 250.0;
+  const double headingOffset = 0.0;
+  const double pitchOffset = 0.0;
+  m_viewshed = new GeoElementViewshed(m_tank, horizontalAngle,
+                                      verticalAngle, minDistance,
+                                      maxDistance, headingOffset,
+                                      pitchOffset, this);
+  m_analysisOverlay->analyses()->append(m_viewshed);
+
+  // Add the Scene to the SceneView
+  m_sceneView->setArcGISScene(scene);
+
+  // create the camera controller to follow the graphic
+  OrbitGeoElementCameraController* followingController = new OrbitGeoElementCameraController(m_tank, 100, this);
+  m_sceneView->setCameraController(followingController);
+
+  // Create a Timer
+  m_timer = new QTimer(this);
+  m_timer->setInterval(100);
+  connect(m_timer, SIGNAL(timeout()), this, SLOT(animate()));
```

Comment:
Can this use the new connection syntax?
`connect(m_timer, &QTimer::timeout, this, &ViewshedGeoElement::animate);`